src/main/java/seedu/address/logic/parser/TimeslotParser.java

```diff
@@ -0,0 +1,261 @@
+
+        String formattedInput = userInput.toUpperCase().trim();
+        String[] dateTimeInputArray = formattedInput.split(REMOVE_WHITESPACE_REGEX);
+        if (formattedInput.contains("NEXT")) {
+            return parseNextDateTime(formattedInput);
+        } else {
+            try {
+                String dateInput = dateTimeInputArray[0];
+                String timeInput = dateTimeInputArray[1];
+
+                StringBuilder parsedFormat = parseFormat(dateInput, timeInput);
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern(String.valueOf(parsedFormat));
+                return LocalDateTime.parse(formattedInput, formatter);
+            } catch (DateTimeParseException e) {
+                throw new ParseException(MESSAGE_INVALID_DATE_TIME_FORMAT);
+            }
+        }
+    }
+
+    /**
+     * Parses a {@code String dateInput, String timeInput} into their appropriate LocalDateTime
+     * format in {@code StringBuilder}. For Absolute DateTime cases only.
+     */
+    public static StringBuilder parseFormat(String dateInput, String timeInput) {
+        StringBuilder formatter = new StringBuilder("");
+
+        if (dateInput.matches(TimeslotRegex.DATE_SLASH_SHORT)) {
+            formatter.append("dd/MM/yy ");
+        } else if (dateInput.matches(TimeslotRegex.DATE_SLASH_LONG)) {
+            formatter.append("dd/MM/yyyy ");
+        } else if (dateInput.matches(TimeslotRegex.DATE_DASH_SHORT)) {
+            formatter.append("dd-MM-yy ");
+        } else {
+            formatter.append("dd-MM-yyyy ");
+        }
+        return parseTimeFormat(timeInput, formatter);
+    }
+
+    /**
+     * Parses a {@code String timeInput, StringBuilder} into a full {@code StringBuilder} format for
+     * LocalDateTime. 24-Hour Clock and Meridian time Inputs are both accepted.
+     */
+    public static StringBuilder parseTimeFormat(String timeInput, StringBuilder formatter) {
+        if (timeInput.contains("AM") || timeInput.contains("PM")) {
+            formatter.append("hh:mma");
+        } else {
+            formatter.append("HH:mm");
+        }
+        return formatter;
+    }
+
+    /**
+     * Parses a {@code String userInput} into a {@code LocalDateTime}.
+     * Parses user input containing the next day, month or year. Time can remain the same or
+     * revised into a new one as per user input.
+     *
+     * @throws ParseException if the given {@code dateTime} does not
+     * conform to the expected date time format.
+     * @throws NullPointerException if the {@code timeInput} does not
+     * conform to the expected time format.
+     */
+    public static LocalDateTime parseNextDateTime(String userInput) throws ParseException, NullPointerException {
+        try {
+            LocalDateTime currentDateTime = LocalDateTime.now().withSecond(0).withNano(0);
+            String[] nextDateTimeInputArray = userInput.split(REMOVE_WHITESPACE_REGEX);
```

Comment:
can try to use `Matcher` instead, check out sebastian's PR on how he uses `Matcher.compile` for `forceDelete` and `patientIndex`. exceptions can then be handled that way.

with this, it probably means you won't get an index out of bounds exception, or nullpointer exception with the input, makes stuff cleaner

Reply:
Hmm aites! I changed it to use matcher instead, and to catch exceptions with a boolean `isInvalidTime` to differentiate the messages likein seb's PR.

But I only created one format with the time -> meaning user has to key in the time and will not be able to leave it empty. Unless I add another conditional for the inputs without time and repeat the `matching for dates` alone, and create another help method to cut short the matching for date. What do yall prefer! 

```
try {
            LocalDateTime currentDateTime = LocalDateTime.now().withSecond(0).withNano(0);
            LocalDateTime parsedDate = null;

            if (nextDateTimeMatcher.matches()) {
                final String nextTimePeriod = nextDateTimeMatcher.group("nextTimePeriod");
                final String timeInput = nextDateTimeMatcher.group("timeInput").trim();

                // need to repeat matching of dates again, because both are separated and process differently for the next command
                if (Arrays.stream(Day.values()).anyMatch(e -> e.name().equals(nextTimePeriod))) {
                    parsedDate = currentDateTime.with(TemporalAdjusters.next(DayOfWeek.valueOf(nextTimePeriod)));
                } else if (nextTimePeriod.contains("MONTH")) {
                    parsedDate = currentDateTime.plusMonths(1);
                } else if (nextTimePeriod.contains("YEAR")) {
                    parsedDate = currentDateTime.plusYears(1);
                } else {
                    isInvalidTime = true;
                }

                int[] hoursMinutesArray = parseTime(timeInput);
                parsedDate = parsedDate.withHour(hoursMinutesArray[0]);
                parsedDate = parsedDate.withMinute(hoursMinutesArray[1]);
                parsedDate = parsedDate.withSecond(0).withNano(0);
            }
```